### PR TITLE
Support `deployPercentage` in `publish` API

### PR DESCRIPTION
**Pull request by remdash**
This PR aims at adding support for the deployPercentage parameter when calling the publish API.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -12,33 +12,33 @@ npm install --save-dev chrome-webstore-upload
...
 ## Usage
 
-All methods return a  promise.
+All methods return a promise.
 
 ### Create a new client
 
 ```javascript
-import chromeWebstoreUpload from 'chrome-webstore-upload';
+import chromeWebstoreUpload from "chrome-webstore-upload";
```
Comment (fregante) [inline readme.md:24]: Please don't make unrelated changes. This was probably Prettier making changes "on save". All of these should be reverted for the PR to be merged

File: index.js [hunk 2]
```diff
@@ -6,8 +6,12 @@ const rootURI = 'https://www.googleapis.com';
 export const refreshTokenURI = 'https://www.googleapis.com/oauth2/v4/token';
 const uploadExistingURI = id =>
     `${rootURI}/upload/chromewebstore/v1.1/items/${id}`;
-const publishURI = (id, target) =>
-    `${rootURI}/chromewebstore/v1.1/items/${id}/publish?publishTarget=${target}`;
+
+const publishURI = (id, target = 'default', deployPercentage) => {
```
Comment (fregante) [inline index.js:10]: See notes in #90 regarding the preferred style (i.e. the object)

Comment (remdash) [reply index.js:10]: Thanks, I will update my PR with those recommendations.

File: readme.md (hunk 1 above)
Comment (remdash) [reply readme.md]: It's fixed

File: index.js (hunk 2 above)
Comment (remdash) [reply index.js]: I've also updated the method to use the URL object as recommended in #90.

File: index.js [hunk 3]
```diff
@@ -61,10 +70,10 @@ class APIClient {
         return response;
     }
 
-    async publish(target = 'default', token = this.fetchToken()) {
+    async publish({ target = 'default', deployPercentage }, token = this.fetchToken()) {
```
Comment (fregante) [inline index.js:73]: Token should be part of it too

File: readme.md [hunk 4]
```diff
@@ -49,7 +49,8 @@ store.uploadExisting(myZipFile, token).then(res => {
 ```javascript
 const target = 'default'; // optional. Can also be 'trustedTesters'
 const token = 'xxxx'; // optional. One will be fetched if not provided
-store.publish(target, token).then(res => {
+const deployPercentage = '25'; // optional. Will default to 100%.
```
Comment (fregante) [inline readme.md:52]: This package should accept a number, not a string

File: index.js [hunk 5]
```diff
@@ -61,10 +70,10 @@ class APIClient {
...
         const { extensionId } = this;
 
-        const request = await fetch(publishURI(extensionId, target), {
+        const request = await fetch(publishURI({ id: extensionId, target, deployPercentage }), {
```
Comment (fregante) [inline index.js:76]: Let's have it accept the full name

```suggestion
        const request = await fetch(publishURI({ extensionId, target, deployPercentage }), {
```

File: readme.md (hunk 4 above)
Comment (remdash) [reply readme.md:52]: it's updated

File: index.js (hunk 3 above)
Comment (remdash) [reply index.js:73]: I've added it in the object

File: index.js (hunk 5 above)
Comment (remdash) [reply index.js:76]: it's fixed too

**Comment by remdash** [comment]
let me know if there is anything else to improve (squashing commits for example).

File: index.js [hunk 6]
```diff
@@ -61,10 +70,10 @@ class APIClient {
         return response;
     }
 
-    async publish(target = 'default', token = this.fetchToken()) {
+    async publish({ target = 'default', deployPercentage, token = this.fetchToken() }) {
```
Comment (fregante) [inline index.js:73]: The function should still work without having to pass `{}`

```suggestion
    async publish({ target = 'default', deployPercentage, token = this.fetchToken() } = {}) {
```

File: readme.md [hunk 7]
```diff
@@ -49,7 +49,8 @@ store.uploadExisting(myZipFile, token).then(res => {
 ```javascript
 const target = 'default'; // optional. Can also be 'trustedTesters'
 const token = 'xxxx'; // optional. One will be fetched if not provided
-store.publish(target, token).then(res => {
+const deployPercentage = 25; // optional. Will default to 100%.
+store.publish({ target, deployPercentage }, token).then((res) => {
```
Comment (fregante) [inline readme.md:53]: The example should be updated too, token is outside

**Review by fregante — APPROVED** [review]

File: readme.md [hunk 8]
```diff
@@ -49,7 +49,8 @@ store.uploadExisting(myZipFile, token).then(res => {
 ```javascript
 const target = 'default'; // optional. Can also be 'trustedTesters'
 const token = 'xxxx'; // optional. One will be fetched if not provided
-store.publish(target, token).then(res => {
+const deployPercentage = 25; // optional. Will default to 100%.
+store.publish({ target, deployPercentage, token }).then((res) => {
```
Comment (fregante) [inline readme.md:53]: On second thought, I'm not ready to release a breaking version. I already have a partial rewrite in progress and I'd want to release _that_ instead. See #84 

Can you revert this API to accept a third parameter?

Comment (remdash) [reply readme.md:53]: So it should be change this way if I understand correctly:
```
store.publish(target, token, deployPercentage)
```

Comment (fregante) [reply readme.md:53]: Yes

Comment (remdash) [reply readme.md:53]: Here it is: https://github.com/fregante/chrome-webstore-upload/pull/93

**Comment by fregante** [comment]
v3.1.0 is out, let me know how it goes

**Comment by fregante** [comment]
v3.1.0 is out, let me know how it goes.

Could you also add a flag to https://github.com/fregante/chrome-webstore-upload-cli?

**Comment by remdash** [comment]
Here is the CLI update MR: https://github.com/fregante/chrome-webstore-upload-cli/pull/83

**Comment by remdash** [comment]
Just a quick comment to tell you that I have had the opportunity to test it and it worked !
